cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/reserve_transfer.rs

```diff
@@ -159,19 +263,58 @@ fn reserve_transfer_native_asset_from_system_para_to_relay_fails() {
 	});
 }
 
+/// Reserve Transfers of native asset from Relay to Parachain should work
+#[test]
+fn reserve_transfer_native_asset_from_relay_to_para() {
+	// Init values for Relay
+	let destination = Westend::child_location_of(PenpalB::para_id());
+	let beneficiary_id = PenpalBReceiver::get();
+	let amount_to_send: Balance = WESTEND_ED * 1000;
+
+	let test_args = TestContext {
+		sender: WestendSender::get(),
+		receiver: PenpalBReceiver::get(),
+		args: relay_test_args(destination, beneficiary_id, amount_to_send),
+	};
+
+	let mut test = RelayToParaTest::new(test_args);
+
+	let sender_balance_before = test.sender.balance;
+	let receiver_balance_before = test.receiver.balance;
+
+	test.set_assertion::<Westend>(relay_to_para_sender_assertions);
+	test.set_assertion::<PenpalB>(para_receiver_assertions);
+	test.set_dispatchable::<Westend>(relay_to_para_limited_reserve_transfer_assets);
+	test.assert();
+
+	let delivery_fees = Westend::execute_with(|| {
+		xcm_helpers::transfer_assets_delivery_fees::<
+			<WestendXcmConfig as xcm_executor::Config>::XcmSender,
+		>(test.args.assets.clone(), 0, test.args.weight_limit, test.args.beneficiary, test.args.dest)
+	});
+
+	let sender_balance_after = test.sender.balance;
+	let receiver_balance_after = test.receiver.balance;
+
+	// Sender's balance is reduced
+	assert_eq!(sender_balance_before - amount_to_send - delivery_fees, sender_balance_after);
+	// Receiver's balance is increased
+	assert!(receiver_balance_after > receiver_balance_before);
```

Comment:
Why not?
```suggestion
	assert_eq!(receiver_balance_after, receiver_balance_before + amount_to_send - delivery_fees);
```

Reply:
because there's also `BuyExecution` actual spent amount which we don't know (even if we did we don't want to hardcode here as weights change)

at most I would tighten this check to `receiver_balance_before < receiver_balance_after && receiver_balance_after < receiver_balance_before + amount_to_send - delivery_fees`